File: NoRPG/Assets/Scripts/Portals/PortalToDesert.js
#pragma strict

function OnTriggerEnter(Col : Collider)
    {
        if (Col.tag == "Player")
        {
            DontDestroyOnLoad(Col.gameObject);
            Application.LoadLevel("Scenes/Desert");
            //SceneManager.LoadScene("Scenes/Desert");
        }
    }

File: NoRPG/Assets/Scripts/Portals/PortalToDesert.cs
using UnityEngine;
using UnityEngine.SceneManagement;

public class PortalToDesert : MonoBehaviour
{
    void OnTriggerEnter()
    {
        PortalControl.control.currentScene = "Desert";

        SceneManager.LoadScene("Scenes/Desert", LoadSceneMode.Single);
    }
}